### ISSUE TITLE
FilterBar has no methods for validation / error reporting

### ISSUE THREAD

**Issue by Stanton** [jadu/pulsar#1144]
The FilterBar current has no methods, styles or documentation for how a field validation error (such as an invalid date) should be handled.

This seems to be a failure of https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html if no additional server-side validation is implemented within products.

> Stanton  3:16 PM
> @jamesjacobs Need to :rubberduck: a behaviour I’m looking at in XFP:
> GIVEN I have a FilterBar with a date filter 
> WHEN I attempt to add an invalid date (e.g. garbage input)
> AND WHEN I attempt to apply this filter
> THEN the filter disappears and no filtering takes place
> I believe this to be a failure of https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html, however… because Pulsar isn’t performing any validation on these fields, this may be a failure on the implementation’s part, however#2, we don’t prescribe any kind of validation styles or error guidelines for filters… Have you done any filter validation in CXM?
> 
> James Jacobs  3:29 PM
> A similar issue exists in CXM, though CXM filters aren’t standard Pulsar filter bar filters (each filter is an form submitted via ajax). It’d been a while since I looked at Pulsar filters but we should totally follow the standard pattern (and use live regions if ajax). Let me remind myself how pulsar filters work :loading:
> 
> James Jacobs  3:36 PM
> ok, this will be fun
> 3:37
> To be honest I’m wondering if technically we even validation here (of course it would be nice)
> 3:38
> Thinking aloud, the user isn’t submitting data (where error identification is critical), they’re effectively submitting search criteria.
> 3:39
> Perhaps if a filter displays no data, then we should display a message (we might already) saying no data found for x, y and z
> 3:40
> Imagine the original form was shown rather than the filter bar, would there be a need to display error warnings if no data was found (or the user searched/filtered for nonsense). (edited) 
> 3:42
> The filter bar just displays a “search” form in a different way
> 3:42
> And a user should be able to search for any thing the form allows, checkbox values, select options, text strings etc
> 3:43
> Perhaps the issue here is just around valid date formats and perhaps email addresses
> 
> Stanton  3:43 PM
> the user isn’t submitting data (where error identification is critical), they’re effectively submitting search criteria.
> I think that’s a very lenient interpretation :smile:
> 
> James Jacobs  3:44 PM
> I think the best way to look at it is by viewing the original form and not the filter bar, what would you expect there
> 3:44
> Feels like a formatting check rather than error validation to me
> 
> Stanton  3:44 PM
> in this case it’s not that the search is returning no data, it’s that the field types are supplied with inputs that “falls outside the required data format or allowed values.”
> 3:44
> and it’s not being caught
> 
> James Jacobs  3:45 PM
> expand on allowed values for me
> 
> Stanton  3:45 PM
> i have a date field
> 3:45
> I give it “BDHJSKABHJ”
> 3:45
> the system isn’t telling me that this value is invalid
> 
> James Jacobs  3:46 PM
> Feels like it should be a JS format checker on the input
> 3:46
> not error validation
> 
> Stanton  3:46 PM
> regardless of the method, we currently have nothing
> 
> James Jacobs  3:46 PM
> Sure, still thinking aloud (edited) 
> 3:47
> It needs sorting just debating the how
> 3:47
> I do something similar in CXM reports with the date filter input, I use moment to check the format… might check other things, would need to check (edited) 
> 
> Stanton  3:47 PM
> something that can be brought over to Pulsar?
> 3:48
> OK, either way if we’re agreed there’s a gap here I’ll ticket it for further thought
> 
> James Jacobs  3:48 PM
> yeah totally
> 
> Stanton  3:48 PM
> Cheers Duck :rubberduck:
> new messages
> 
> James Jacobs  3:50 PM
> Not something that can be copied over as there’s lots of custom unrelated things going on but essentially the check is just confirming it’s a valid date and format
> 3:51
> toggleSubmitButton ($submitButton, $fakeStartDate, $fakeEndDate, format) {
>    if (this.moment($fakeStartDate.val(), format, true).isValid() && this.moment($fakeEndDate.val(), format, true).isValid()) {
>        $submitButton.removeAttr('disabled');
>    } else {
>        $submitButton.attr('disabled', true);
>    }
> }
